### PR TITLE
Code block component

### LINKED ISSUE #232: pre > code background color is wrong on IE11
The problem is related with the [`pre code` background property](https://github.com/bitnami/hex/blob/14f90f8f0541197e8684a8c22b33711140d9bb1e/packages/hex-core/src/core/_code.scss#L33):

```
pre code {
  background-color: initial;
  // ...
}
``` 

`initial` nor `unset` are valid values in IE, so it's rendering the `code` background property making the text illegible. We need to use the `inherit` value.

**Ref**: https://alligator.io/css/inherit-initial-unset/

screen shot 2018-09-20 at 15 36 03

### LINKED ISSUE #239: Pre tags overflows inner content
`pre` tags must keep all the content under scroll.

screenshot 2018-10-15 at 08 48 56

### PR DESCRIPTION
**Pull request by Angelmmiguel**
`Code` component highlights code snippets using [highlight.js](https://highlightjs.org). By default, it doesn't include any language so you need to follow the instructions to register a new language.

Another interesting feature is that it fixes the indentation issues with `pre` tags and JSX. However, all these kind of features can be disabled using props.

screenshot 2018-10-15 at 08 56 32

References #219. It closes #239. It closes #232 

PS: The documentation itself uses `Code` to display some code snippets 😂

### CODE REVIEW

File: packages/hex-react/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module.exports = api => {
```
Comment (Angelmmiguel) [inline packages/hex-react/babel.config.js]: I had to move the configuration of babel to this file so it's used in all the project.

File: packages/hex-react/webpack.common.js [hunk 2]
```diff
@@ -18,6 +18,15 @@ module.exports = {
         exclude: /(node_modules)/,
         loader: ['babel-loader', 'eslint-loader'],
       },
+      {
+        test: /\.js$/,
+        include: path.resolve(__dirname, './node_modules/detect-indent'),
```
Comment (Angelmmiguel) [inline packages/hex-react/webpack.common.js]: This module is not transpiled and it was causing some issues in IE11.

**Comment by Angelmmiguel** [comment]
I was facing some issues with the `detect-indent` package in IE 11 because it's not properly transpiled. For that reason, I switched the babel config to [`babel.config.js` (project-wide)](https://babeljs.io/docs/en/configuration#what-s-your-use-case) and include the library in the webpack babel processing.

File: packages/hex-docz/examples/RegisterLanguage.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { Code } from '@bitnami/hex-react/src'
+
+import js from 'highlight.js/lib/languages/javascript';
+Code.registerLanguage('js', js);
+
+const RegisterLanguage = () => (
+  <Code>
+    {`
+      // Components
+      import { Code } from '@bitnami/hex-react';
```
Comment (andresmgot) [inline packages/hex-docz/examples/RegisterLanguage.js]: typo? `import { Code } from '@bitnami/hex-react/src'`

Comment (Angelmmiguel) [reply packages/hex-docz/examples/RegisterLanguage.js]: In this case it's fine. It's just an example about how to include `Code` in a normal project. I'm importing from `src` in the MDX files to be able to read the `props` in `PropsTable`

**Pull request merged**